code/framework/api/src/main/java/io/cattle/platform/api/servlet/ApiRequestFilter.java

```diff
@@ -200,7 +204,13 @@ private void addPLCookie(HttpServletRequest httpRequest, HttpServletResponse res
         }
 
         if (plCookie == null || !PL_SETTING.getValue().equalsIgnoreCase(plCookie.getValue())) {
-            plCookie = new Cookie(PL, PL_SETTING.getValue());
+            String plValue = PL_SETTING.getValue();
+            try {
+              plValue = URLEncoder.encode(PL_SETTING.getValue(), "UTF-8");
+            } catch (IOException e) {
+              log.error("Failed to encode PL value with UTF-8.", e);
```

Comment:
should we return when exception happens? the cookie value would be nil if exception happens

Reply:
added. Thanks

Reply:
added